src/main/java/ua/controller/ProductController.java

```diff
@@ -1,4 +1,48 @@
 package ua.controller;
 
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.web.bind.annotation.*;
+import ua.domain.Product;
+import ua.service.ProductService;
+
+import java.util.List;
+
+@RestController
+@RequestMapping(value = "/products", produces = MediaType.APPLICATION_JSON_VALUE)
 public class ProductController {
-}
+
+    private final ProductService productService;
+
+    public ProductController(ProductService productService) {
+        this.productService = productService;
+    }
+
+    @GetMapping
+    public List<Product> getProducts() {
+        return productService.getAll();
+    }
+
+    @GetMapping(value = "/{id}")
+    public Product getOneProduct(@PathVariable Long id) {
+        return productService.getEntry(id);
+    }
+
+    @RequestMapping(method = RequestMethod.POST)
+    @ResponseStatus(HttpStatus.CREATED)
+    public Product createProduct(@RequestBody Product product) {
+        return productService.addEntry(product);
+    }
+
+    @DeleteMapping(value = "/{id}")
+    @ResponseStatus(HttpStatus.NO_CONTENT)
+    public void deleteProduct(@PathVariable Long id) {
+        productService.delete(id);
+    }
+
+    @RequestMapping(method = RequestMethod.PUT)
+    @ResponseStatus(HttpStatus.CREATED)
+    public Product updateProduct(Long id, Product product) {
```

Comment:
id should be marked as @PathVariable, and product as @RequestBody. And also in RequestMapping value {id} is missed